tools/archives/pyvo_integration/astronomical_archives.xml

```diff
@@ -0,0 +1,977 @@
+              <param name="table" type="text" label="Table name" help="Name of the table you want to query (FROM clause)" />
+              <section name="where_clause" title="WHERE clause (field = value)" expanded="false">
+                <param name="where_field" type="text" label="WHERE field" help="WHERE **field** = value" />
+                <param name="where_condition" type="text" label="WHERE value" help="WHERE field = **value**" />
+              </section>
+              <param name="url_field" type="text" label="Url field" help="Table field containing the url of the file to download" />
+            </when>
+          </conditional>
+        </section>
+        <section name="output_section" title="Output selection" expanded="true">
+          <param name="number_of_files" type="integer" value="1" min="1" max="100" label="Number of files or urls to download" help="Beware of disk space usage when downloading large number of files!" />
+          <param name="output_selection" type="select" label="Get" multiple="true" display="checkboxes">
+            <option selected="true" value="c">Return URL list as CSV</option>
+            <option value="i">Download files</option>
+            <option value="h">Return URL list in extended HTML (requires HTML rendering permission, see help)</option>
+            <option value="b">Return URL list as HTML</option>
+          </param>
+        </section>
+
+    </inputs>
+    <outputs>
+        <data name="output" format="fits" label="${tool.name} -> File from ${archive_selection.archive_type} search:" >
+          <discover_datasets pattern="__designation_and_ext__" directory="fits" visible="true" format="fits" />
+          <filter>'i' in output_section['output_selection']</filter>
+          <filter>output_section['output_selection'] is not None</filter>
+        </data>
+        <data name="output_csv" format="txt" label="${tool.name} -> CSV Manifest:">
+          <filter>'c' in output_section['output_selection']</filter>
+          <filter>output_section['output_selection'] is not None</filter>
+        </data>
+        <data name="output_html" format="html" label="${tool.name} -> HTML Preview:">
+          <filter>'h' in output_section['output_selection']</filter>
+          <filter>output_section['output_selection'] is not None</filter>
+        </data>
+        <data name="output_basic_html" format="html" label="${tool.name} -> Basic HTML Preview:">
+          <filter>'b' in output_section['output_selection']</filter>
+          <filter>output_section['output_selection'] is not None</filter>
+        </data>
+        <data name="output_error" format="txt" label="${tool.name} -> Query Summary:"></data>
+    </outputs>
+    <tests>
+        <test expect_num_outputs="1">
+            <param name="output_selection" value="i"/>
+            <param name="number_of_files" value="1"/>
+            <conditional name="archive_selection">
+                <param name="archive_type" value="registry"/>
```

Comment:
it would be nice to have at least one test using the data_table.

Reply:
I had some when I was using "from file" instead of a data_table but when running planemo test I always got some "invalid value" or something like that.

I could try it again now that we moved to data_table

Reply:
maybe that one helps you https://github.com/galaxyproject/tools-iuc/tree/96f8a533278b4b6394aebd7a8f537513b0d29b1a/tools/recentrifuge